homeassistant/components/netatmo/light.py

```diff
@@ -0,0 +1,145 @@
+"""Support for the Netatmo camera lights."""
+import logging
+
+import pyatmo
+
+from homeassistant.components.light import LightEntity
+from homeassistant.core import callback
+from homeassistant.exceptions import PlatformNotReady
+
+from .const import DATA_HANDLER, DOMAIN, MANUFACTURER, SIGNAL_NAME
+from .data_handler import CAMERA_DATA_CLASS_NAME, NetatmoDataHandler
+from .netatmo_entity_base import NetatmoBase
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Netatmo camera light platform."""
+    if "access_camera" not in entry.data["token"]["scope"]:
+        _LOGGER.info(
+            "Cameras are currently not supported with this authentication method"
+        )
+        return
+
+    data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
+
+    async def get_entities():
+        """Retrieve Netatmo entities."""
+        await data_handler.register_data_class(
+            CAMERA_DATA_CLASS_NAME, CAMERA_DATA_CLASS_NAME, None
+        )
+
+        entities = []
+        try:
+            all_cameras = []
+            for home in data_handler.data[CAMERA_DATA_CLASS_NAME].cameras.values():
+                for camera in home.values():
+                    all_cameras.append(camera)
+
+            for camera in all_cameras:
+                if camera["type"] == "NOC":
+                    if not data_handler.webhook:
+                        raise PlatformNotReady
+
+                    _LOGGER.debug(
+                        "Adding camera light %s %s", camera["id"], camera["name"]
+                    )
+                    entities.append(
+                        NetatmoLight(
+                            data_handler,
+                            camera["id"],
+                            camera["type"],
+                            camera["home_id"],
+                        )
+                    )
+
+        except pyatmo.NoDevice:
```

Comment:
What line can raise here? This try... except block looks bigger than neeed. We want that scope to be as small as possible.